doc-snippets/api/src/main/java/otel/AsyncCounterUsage.java

```diff
@@ -0,0 +1,50 @@
+package otel;
+
+import static otel.Util.WIDGET_COLOR;
+import static otel.Util.WIDGET_SHAPE;
+import static otel.Util.computeWidgetColor;
+import static otel.Util.computeWidgetShape;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.metrics.ObservableLongCounter;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class AsyncCounterUsage {
+  // Pre-allocate attributes whenever possible
+  private static final Attributes WIDGET_RED_CIRCLE = Util.WIDGET_RED_CIRCLE;
+
+  public static void asyncCounterUsage(Meter meter) {
+    AtomicLong widgetCount = new AtomicLong();
+
+    // Construct an async counter to observe an existing counter in a callback
+    ObservableLongCounter asyncCounter =
+        meter
+            .counterBuilder("fully.qualified.counter")
+            .setDescription("A count of produced widgets")
+            .setUnit("{widget}")
+            // optionally change the type to double
+            // .ofDoubles()
```

Comment:
```suggestion
            // type can be changed to double by calling .ofDoubles()
```

Reply:
makes sense, what do you think of this?

```suggestion
            // uncomment to optionally change the type to double
            // .ofDoubles()
```

Reply:
I like it. I'll update it elsewhere too.